Classes/YTPlayerView.m

```diff
@@ -825,13 +825,44 @@ - (NSString *)stringFromVideoIdArray:(NSArray *)videoIds {
 }
 
 /**
- * Private method for evaluating JavaScript in the WebView.
+ * Private method for evaluating JavaScript in the webview.
  *
  * @param jsToExecute The JavaScript code in string format that we want to execute.
- * @return JavaScript response from evaluating code.
  */
-- (NSString *)stringFromEvaluatingJavaScript:(NSString *)jsToExecute {
-  return [self.webView stringByEvaluatingJavaScriptFromString:jsToExecute];
+- (void)stringFromEvaluatingJavaScript:(NSString *)jsToExecute {
+  [self stringFromEvaluatingJavaScript:jsToExecute completionHandler:nil];
+}
+
+/**
+ * Private method for evaluating JavaScript in the webview.
+ *
+ * @param jsToExecute The JavaScript code in string format that we want to execute.
+ * @param completionHandler A block to invoke when script evaluation completes or fails.
+ */
+- (void)stringFromEvaluatingJavaScript:(NSString *)jsToExecute
+                     completionHandler:(_Nullable YTStringCompletionHandler)completionHandler {
+  [_webView evaluateJavaScript:jsToExecute
+             completionHandler:^(id _Nullable result, NSError *_Nullable error) {
+    if (!completionHandler) {
+      return;
+    }
+
+    if (error) {
+      completionHandler(nil, error);
+      return;
+    }
+    if (![result isKindOfClass:[NSString class]]) {
```

Comment:
Currently, with all these changes implemented, some functions return **double** value which results in failing this command. E.x. player.getCurrentDuration();

Reply:
fixed this, thanks for the feeback.